src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/commands.py

```diff
@@ -216,6 +231,27 @@
                                      custom_func_name='server_ad_admin_update',
                                      setter_arg_name='properties')
 
+    server_keys_operations = create_service_adapter('azure.mgmt.sql.operations.server_keys_operations',
+                                                    'ServerKeysOperations')
+
+    with ServiceGroup(__name__, get_sql_server_keys_operations, server_keys_operations, custom_path) as s:
+        with s.group('sql server key') as c:
+            c.custom_command('create', 'server_key_create')
+            c.custom_command('delete', 'server_key_delete')
+            c.custom_command('show', 'server_key_get')
+            c.command('list', 'list_by_server')
+
+    encryption_protectors_operations = create_service_adapter(
+        'azure.mgmt.sql.operations.encryption_protectors_operations',
+        'EncryptionProtectorsOperations')
+
+    with ServiceGroup(__name__, get_sql_encryption_protectors_operations, encryption_protectors_operations,
+                      custom_path) as s:
+        with s.group('sql server encryption-protector') as c:
+            c.command('show', 'get')
+            c.command('list', 'list_by_server')
+            c.custom_command('update', 'encryption_protector_update')
```

Comment:
I didn't see the implementation of this, but since there is not a `create` command, it leads me to believe this should be called `set`.  The distinction:

`create`: traditional ARM PUT.  Effectively a create or replace.
`update`: uses an ARM PATCH or simulates PATCH behind the scenes. If it has a "create or patch" behavior, then this is the appropriate verb too.
`set`: Basically a PUT that is used in the absence of separate create and update commands.

Reply:
@mihymel is it possible that there might be some updateable properties and therefore might want to have an `update` command in the future? If yes, to me it makes more sense to have an `update` command now rather than `set` now and also `update` later. But you might disagree, just my 2c

Reply:
You can think of the encryption protector just as "the server key that will be used for transparent data encryption". With this in mind, you can understand that encryption protector probably won't need updateable properties in the future. I think `set` makes a lot of sense.

Reply:
Ok. That sounds fine. Are `show` and `list_by_server` both needed? What happens if you `show` on a fresh server that you have never `set` before?

Reply:
Just brainstorming here: what do you think about commands like `sql db tde set-key` , `sql db tde show-key` ?